### PR TITLE
Allow move calculations for all files

### PR DESCRIPTION
**Pull request by egibs**
Given how slow move calculations can be, only `.so` library files have been considered when running bincapz with the `--diff` flag. 

This PR allows for all files to be considered during move calculations and addresses the `missing`/`added` output seen here: https://github.com/wolfi-dev/os/pull/23820

I added a quick diff test for this case and the output is now:
```
*** changed: Linux/clean/aws-c-io/aws-c-io-0.14.11-r0.spdx.json
```

This PR adds a function that flattens the diff report maps into a single slice so that iterating through them and calculating scores is faster.

### CODE REVIEW

**Review by hectorj2f — APPROVED** [review]

File: samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff
Comment (tstromberg) [inline samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff:9]: This should be a file change, not a add/removal.

File: samples/macOS/clean/ls.mdiff
Comment (tstromberg) [inline samples/macOS/clean/ls.mdiff:1]: This should be a file change - but instead of `.`, it should say the name of the new filename.

Comment (egibs) [reply samples/macOS/clean/ls.mdiff:1]: Addressed in [`dfc5173` (#356)](https://github.com/chainguard-dev/bincapz/pull/356/commits/dfc517390378280a98a62403b6d9c29f3d6075e3).

**Review by tstromberg — APPROVED** [review]

**Comment by egibs** [comment]
Still working on the `moved` logic for the new sample but once that's done I'll merge this.

File: samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff
Comment (tstromberg) [inline samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff:1]: When diffing two binaries, it shouldn't be considered a move; just a diff between two programs. 'Moved' should only be considered if directory trees are diffed.

File: pkg/action/diff.go
Comment (egibs) [inline pkg/action/diff.go:243]: I added this because running something like this makes bincapz think the original binary was changed:
```sh
$ go run . --format=simple --diff samples/Linux/clean/ls.x86_64 samples/macOS/2024.LightSpy/dropper
```

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff
Comment (egibs) [reply samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff:1]: Makes sense.

I'll land the original changes as-is and then spend some more time overhauling the diff behavior. I don't want this scope creep to hold up the next release.

Comment (egibs) [reply samples/Linux/2023.FreeDownloadManager/freedownloadmanager.sdiff]: Moved back to `filepath.Rel` in [`55a9ed2` (#356)](https://github.com/chainguard-dev/bincapz/pull/356/commits/55a9ed242264ffefb9ab919d59fac16d4bde14ae).

**Review by tstromberg — APPROVED** [review]

**Pull request merged**
